Make syntax highlighting more similar to Rust

sdankel opened the pull request:
Closes https://github.com/FuelLabs/sway-vscode-plugin/issues/72

This is making sway's tmLanguage grammar nearly identical to the [rust grammar](https://raw.githubusercontent.com/dustypomerleau/rust-syntax/master/syntaxes/rust.tmLanguage.json).

The differences I added for sway are:
- `abi` is a special keyword
- `mod` in rust -> `dep` in sway
- `predicate`, `library`, `script`, and `contract` match the `#[]` highlighting in rust ("meta attribute")
- `asm` is highlighted as a meta attribute

Left: this branch, Right: master
<img width="1037" alt="image" src="https://user-images.githubusercontent.com/47993817/184286907-c3f8d383-d09d-4bf2-80b9-66d94a52dbb7.png">

Left: sway, Right: rust, Theme: Pale Fire Dark
<img width="1204" alt="image" src="https://user-images.githubusercontent.com/47993817/184286389-2accc4ba-5d03-4f86-abdb-7b3bd1eb4290.png">

Left: sway, Right: rust, Theme: Dark+
<img width="1201" alt="image" src="https://user-images.githubusercontent.com/47993817/184286476-34d694cd-fcdb-4eb2-a1d5-b83e9780b6e1.png">

asm keyword:
<img width="1083" alt="image" src="https://user-images.githubusercontent.com/47993817/184294095-b991aeac-3f50-4ccb-9cc3-3ff9e750a1ee.png">

JoshuaBatty commented:
Thanks @sdankel. Just pulled down and tested locally with a few different color palettes. Still seems like there are some inconsistencies between Rust and Sway. 

I did notice that the similarities are almost nonexistent using the default Dark+ theme.

Here are 2 examples, Sway on the left, Rust on the right. The first is using Palefire theme while the second is using the Monokai theme.

![langauge comparison](https://user-images.githubusercontent.com/1289413/183787071-189bc860-026c-4292-bf4e-ac88fb151de4.jpg)

![langauge comparison 2](https://user-images.githubusercontent.com/1289413/183787082-ad05bbec-6185-4d2a-8387-fc6d56a70e2b.jpg)

2 things that are of note, the "struct" keyword is colored differently, and the "let" keywords are also different. There are a few differences in the fn new function but don't worry about them yet as I'm pretty sure we can address those in the language server.

Would it be possible to take a look at getting the keywords matching before we land this PR?

sdankel commented:
> 2 things that are of note, the "struct" keyword is colored differently, and the "let" keywords are also different. There are a few differences in the fn new function but don't worry about them yet as I'm pretty sure we can address those in the language server.
> 
> Would it be possible to take a look at getting the keywords matching before we land this PR?

Updated `enum`, `struct`, `let`, and `const` to match rust, and updated the screenshots in the description.

eureka-cpu commented on client/test/syntaxes/basic_predicate.sw.snap:
```diff
@@ -0,0 +1,22 @@
+#    ^^^^ source.sway constant.language.bool.sway
+>}
+#^ source.sway punctuation.brackets.curly.sway
+>
```
GitHub will complain about missing new lines at the end of files

sdankel replied:
These are generated files, so we shouldn't modify them by hand. I think they're ignored by the linter since the file extension is `.snap`

eureka-cpu replied:
ah, makes sense! in that case, disregard my previous comments haha

eureka-cpu commented on client/test/syntaxes/chained_if_let.sw.snap:
```diff
@@ -0,0 +1,180 @@
+#    ^ source.sway punctuation.brackets.curly.sway
+>}
+#^ source.sway punctuation.brackets.curly.sway
+>
```
Same here

eureka-cpu commented on client/test/syntaxes/const_decl.sw.snap:
```diff
@@ -1,34 +1,49 @@
 >}
-#^^ source.sway
+#^ source.sway punctuation.brackets.curly.sway
 >
```
Here

eureka-cpu commented on client/test/syntaxes/match_expressions_mismatched.sw.snap:
```diff
@@ -0,0 +1,164 @@
+#    ^ source.sway punctuation.brackets.curly.sway
+>}
+#^ source.sway punctuation.brackets.curly.sway
+>
```
^

eureka-cpu commented on client/test/syntaxes/smo_opcode.sw.snap:
```diff
@@ -0,0 +1,76 @@
+#    ^^^^ source.sway constant.language.bool.sway
+>}
+#^ source.sway punctuation.brackets.curly.sway
+>
```
^

eureka-cpu commented on client/test/syntaxes/storage_declaration.sw.snap:
```diff
@@ -0,0 +1,58 @@
+#          ^ source.sway meta.function.definition.sway punctuation.brackets.curly.sway
+>}
+#^ source.sway punctuation.brackets.curly.sway
+>
```
last one

eureka-cpu reviewed (approved):
utACK

JoshuaBatty reviewed (approved):
Thanks for making those changes @sdankel, looking good!
